HeapTracker/README.md

```diff
@@ -0,0 +1,120 @@
+
+    ```c
+    Remote debugging from host 192.168.35.1, port 60851
+    Starting Heap Tracker test application...
+    consumeHeap_malloc --> Heap status: max available(256000 bytes), allocated (0 bytes)
+    Heap-Tracker lib: malloc(1024)=0xbeee9010... SUCCESS: heap_allocated (1024 bytes) - delta with heap_threshold(254976 bytes)
+    Heap-Tracker lib: _free(0xbeee9010,1024)... SUCCESS: heap_allocated (0 bytes) - delta with heap_threshold(256000 bytes)
+    Heap-Tracker lib: malloc(2048)=0xbee7b3e0... SUCCESS: heap_allocated (2048 bytes) - delta with heap_threshold(253952 bytes)
+    ...
+    ...
+    Heap-Tracker lib: _free(0xbedcf010,256000)... SUCCESS: heap_allocated (0 bytes) - delta with heap_threshold(256000 bytes)
+    Heap-Tracker lib: malloc(257024)=0xbedcf010... WARNING: heap_allocated (257024 bytes) is above heap_threshold (256000 bytes)
+    Heap-Tracker lib: _free(0xbedcf010,257024)... SUCCESS: heap_allocated (0 bytes) - delta with heap_threshold(256000 bytes)
+    consumeHeap_malloc --> Currently available heap up to given heap_threshold: 251Kb (257024 bytes)
+    ```
+
+- On failure, the OS's OOM Killer will SIGKILL the App's process. This will be the indicator that the global variable `heap_threshold` in `heap_tracker_lib.c` should be lowered to a value below to the last successful allocation:
+
+    ```c
+    Remote debugging from host 192.168.35.1, port 60827
+    Starting Heap Tracker test application...
+    consumeHeap_realloc --> Heap status: max available(256000 bytes), allocated (0 bytes)
+    Heap-Tracker lib: _realloc(0,0,1024)=0xbeee9010... SUCCESS: heap_allocated (1025 bytes) - delta with heap_threshold(254975 bytes)
+    Heap-Tracker lib: _realloc(0xbeee9010,1024,2048)=0xbeee9010... SUCCESS: heap_allocated (2050 bytes) - delta with heap_threshold(253950 bytes)
+    Heap-Tracker lib: _realloc(0xbeee9010,2048,3072)=0xbeee9010... SUCCESS: heap_allocated (3075 bytes) - delta with heap_threshold(252925 bytes)
+    ...
+    ...
+    Heap-Tracker lib: _realloc(0xbeeff010,189440,190464)=0xbeeff010... SUCCESS: heap_allocated (190650 bytes) - delta with heap_threshold(65350 bytes)
+    Heap-Tracker lib: _realloc(0xbeeff010,190464,191488)=0xbeeff010... SUCCESS: heap_allocated (191675 bytes) - delta with heap_threshold(64325 bytes)
+
+    Child terminated with signal = 0x9 (SIGKILL)
+    ```
+
+## Key concepts
+The goal of the Heap Tracker library, is to support developers track their High-level application's memory requests to match the expected behavior throughout the application execution time (i.e. a constant raise in value of `heap_allocated` may indicate a potential memory leak).
+
+For more information and recommendations on Azure Sphere memory usage and debugging, refer to [Memory use in high-level applications](https://docs.microsoft.com/en-us/azure-sphere/app-development/application-memory-usage).
+
+## Next steps
```

Comment:
[nit] this heading is odd, the next two sections arent really "next steps" -- suggest removing this heading and promoting the next 2 sections to ##

Reply:
Agree. I just conformed to the gallery template: @tylerfox @rebeccaleigh, how could we best redact?

Reply:
looks good now, thanks!